dart/biomechanics/MarkerFitter.cpp

```diff
@@ -710,16 +714,37 @@ MarkerFitter::MarkerFitter(
             * (-1 * this->mAnthropometricWeight);
       for (int i = 0; i < this->mSkeleton->getNumBodyNodes(); i++)
       {
-        state->bodyScalesGrad.col(i) = bodyScalesGradVector.segment<3>(i * 3);
+        state->bodyScalesGrad.col(i) += bodyScalesGradVector.segment<3>(i * 3);
       }
       // 3.4. Reset
       this->mSkeleton->setBodyScales(oldBodyScales);
     }
-    else
+    loss += anthroRegularization;
+
+    // 4. Regularize the height of the body, if we've got a height prior
+    s_t heightRegularization = 0.0;
+    if (mHeightPriorWeight > 0.0)
     {
-      state->bodyScalesGrad.setZero();
+      Eigen::VectorXs pose
+          = Eigen::VectorXs::Zero(this->mSkeleton->getNumDofs());
+      pose(4) = 0.94;
```

Comment:
This seems to be hard-coding the default y-position of the model? Will the loss term not function correctly if a generic model with a different y-pos convention is used?

Reply:
This is an excellent catch. I was just blindly copying the way the `ANSUR_metrics.xml` defines the height metric, but that's dumb. It shouldn't matter what the root translation of the model is when measuring height, so I've deleted these offset lines.